File: documents/aws-doc-sdk-examples/php/example_code/sqs/GetQueueUrl.php
<?php
/**
 * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * This file is licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License. A copy of
 * the License is located at
 *
 * http://aws.amazon.com/apache2.0/
 *
 * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 *  ABOUT THIS PHP SAMPLE: This sample is part of the SDK for PHP Developer Guide topic at
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/sqs-examples-using-queues.html
 *
 */
// snippet-start:[sqs.php.get_queue_url.complete]
// snippet-start:[sqs.php.get_queue_url.import]
require 'vendor/autoload.php';

use Aws\Sqs\SqsClient; 
use Aws\Exception\AwsException;
// snippet-end:[sqs.php.get_queue_url.import]

/**
 * Get SQS queue url
 *
 * This code expects that you have AWS credentials set up per:
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html
 */
// snippet-start:[sqs.php.get_queue_url.main]

$queueName = "SQS_QUEUE_NAME";
 
$client = new SqsClient([
    'profile' => 'default',
    'region' => 'us-west-2',
    'version' => '2012-11-05'
]);

try {
    $result = $client->getQueueUrl([
        'QueueName' => $queueName // REQUIRED
    ]);
    var_dump($result);
} catch (AwsException $e) {
    // output error message if fails
    error_log($e->getMessage());
}
 
 
// snippet-end:[sqs.php.get_queue_url.main]
// snippet-end:[sqs.php.get_queue_url.complete]
// snippet-comment:[These are tags for the AWS doc team's sample catalog. Do not remove.]
// snippet-sourcedescription:[GetQueueUrl.php demonstrates how to return the URL of an existing Amazon SQS queue.]
// snippet-keyword:[PHP]
// snippet-sourcesyntax:[php]
// snippet-keyword:[AWS SDK for PHP v3]
// snippet-keyword:[Code Sample]
// snippet-keyword:[Amazon Simple Queue Service]
// snippet-service:[sqs]
// snippet-sourcetype:[full-example]
// snippet-sourcedate:[2018-12-27]
// snippet-sourceauthor:[jschwarzwalder (AWS)]

File: documents/aws-doc-sdk-examples/cpp/example_code/sqs/get_queue_url.cpp
 
//snippet-sourcedescription:[get_queue_url.cpp demonstrates how to retrieve the URL of an Amazon SQS queue.]
//snippet-keyword:[C++]
//snippet-sourcesyntax:[cpp]
//snippet-keyword:[Code Sample]
//snippet-keyword:[Amazon Simple Queue Service]
//snippet-service:[sqs]
//snippet-sourcetype:[full-example]
//snippet-sourcedate:[]
//snippet-sourceauthor:[AWS]


/*
   Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.

   This file is licensed under the Apache License, Version 2.0 (the "License").
   You may not use this file except in compliance with the License. A copy of
   the License is located at

    http://aws.amazon.com/apache2.0/

   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
   CONDITIONS OF ANY KIND, either express or implied. See the License for the
   specific language governing permissions and limitations under the License.
*/
//snippet-start:[sqs.cpp.get_queue_url.inc]
#include <aws/core/Aws.h>
#include <aws/sqs/SQSClient.h>
#include <aws/sqs/model/GetQueueUrlRequest.h>
#include <aws/sqs/model/GetQueueUrlResult.h>
#include <iostream>
//snippet-end:[sqs.cpp.get_queue_url.inc]

/**
 * Gets the url associated with an sqs queue based on command line input
 */
int main(int argc, char** argv)
{
    if (argc != 2) {
        std::cout << "Usage: get_queue_url <queue_name>" << std::endl;
        return 1;
    }

    Aws::SDKOptions options;
    Aws::InitAPI(options);
    {
        Aws::String queue_name = argv[1];

        // snippet-start:[sqs.cpp.get_queue_url.code]
        Aws::SQS::SQSClient sqs;

        Aws::SQS::Model::GetQueueUrlRequest gqu_req;
        gqu_req.SetQueueName(queue_name);

        auto gqu_out = sqs.GetQueueUrl(gqu_req);
        if (gqu_out.IsSuccess()) {
            std::cout << "Queue " << queue_name << " has url " <<
            gqu_out.GetResult().GetQueueUrl() << std::endl;
        } else {
            std::cout << "Error getting url for queue " << queue_name << ": " <<
                gqu_out.GetError().GetMessage() << std::endl;
        }
        // snippet-end:[sqs.cpp.get_queue_url.code]
    }
    Aws::ShutdownAPI(options);
    return 0;
}

